enumerable_methods.rb

```diff
@@ -123,14 +134,16 @@ def my_count
   end
 
   def my_inject
-    return to_enum unless block_given?
-
     arr = self
     if arr.is_a? Array
       x = arr[0]
       i = 1
       while i < arr.length
-        x = yield(x, arr[i])
+        x = if block_given?
+              yield(x, arr[i])
+            else
+              [x, arr[i]]
+            end
         i += 1
       end
```

Comment:
**[REQUIRED]**
* Combines all elements of enum by applying a binary operation, specified by a block.
```
Failure/Error: actual = range.my_inject(4) { |prod, n| prod * n }
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
```
* When a symbol is specified combines each element of the collection by applying the symbol as a named method.
```
Failure/Error: expect(array.my_inject(:+)).to eq array.inject(:+)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
```